### DISCUSSION TITLE
php file_get_content return 404

### DISCUSSION THREAD

**Discussion by yo57810** [osm-search/Nominatim#2960] [Q&A]
Impossible d'appeler l'API à partir d'un site web hébergé chez OVH : je reçoit une erreur 404 !

Code PHP
$url="https://nominatim.openstreetmap.org/search?street=8%2C+rue+du+Ch%C3%A2teau&city=RECHICOURT-LE-CHATEAU&country=&postalcode=&format=geojson";
file_get_contents("$url");

idem avec 
$url = "https://nominatim.openstreetmap.org";

Test : https://www.biblebiere.com/Test_API_Nominatim.php

**Comment by lonvia** [comment]
If you get a 404, then something is wrong with the URL string.

If you get a 403 or 429, then please read our usage policy here: https://operations.osmfoundation.org/policies/nominatim/ and look at the answer to this question: https://help.openstreetmap.org/questions/59788/calling-nominatim-with-file_get_contents

**Reply by yo57810** [reply]
Hi, thank you, it works now ;
i had to add this :

// Create a stream
$opts = array('http'=>array('header'=>"User-Agent: StevesCleverAddressScript 3.7.6\r\n"));
$context = stream_context_create($opts);

file_get_contents("$url", false, $context);
